src/execution/join_hashtable.cpp

```diff
@@ -585,14 +595,27 @@ void ScanStructure::ConstructMarkJoinResult(DataChunk &join_keys, DataChunk &chi
 			}
 		}
 	}
+#ifdef LINEAGE
+	unique_ptr<uintptr_t[]> key_locations_lineage(new uintptr_t[child.size()]);
+	auto ptrs = FlatVector::GetData<uintptr_t>(this->pointers);
+#endif
 	// now set the remaining entries to either true or false based on whether a match was found
 	if (found_match) {
 		for (idx_t i = 0; i < child.size(); i++) {
 			bool_result[i] = found_match[i];
+#ifdef LINEAGE
+			key_locations_lineage[i] = found_match[i];
```

Comment:
if we want to keep only bool_result we wouldn't know which value from the build side was matched with. We would need to keep both prts and bool_result as well. it would be cheaper to persist but we need to adjust the interface for the binary operator